### ISSUE TITLE
[Bug]-[12]:RegEx Validation Not Working in Input Field of JSON Form

### ISSUE THREAD

**Issue by ame-appsmith** [appsmithorg/appsmith#14109]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

In a **JSON Form**, when using **Regex** on an **Input** field, the field shows as _invalid_, even if the value entered is correct.

![image](https://user-images.githubusercontent.com/101268423/170367427-cb702add-0868-4026-939c-1d5928081dde.png)

### Steps To Reproduce

1. Add a **JSON Form** to the canvas and enter some **Source Data**. E.g.:
```
{
  "name": "John",
  "date_of_birth": "20/02/1990",
  "employee_id": 1001,
  "phone": "1234567890"
}
```
2. In the **Regex** property of the _Phone_ input field, enter `^\d+$` to allow only digits.
3. Type some digits in the _Phone_ input field.

**Expected behavior**: The field should be valid, as only digits have been entered.
**Actual behavior**: The field is shown as _invalid_.

### Public Sample App

https://app.appsmith.com/app/json-form/json-form-regex-validation-628e95f77901344ba8d2b2b1

### Version

Cloud  v1.6.26-SNAPSHOT

**Comment by dilippitchika** [comment]
## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 3 |
| Effort (months)  | 0.5 |

**Comment by ginjojohny** [comment]
I am facing the same issue as mentioned above in my application . Validation using RegEx not working properly. While entering the value in input field , it takes some time for validating the input and showing the input as invalid even if it is a valid one.

![Screenshot (147)](https://user-images.githubusercontent.com/94359116/177981690-26b23620-4c22-446c-abd3-1db98d40b1aa.png)

**Comment by ginjojohny** [comment]
I am facing the same issue in my application. Validation using Regex is not working properly. While we enter value in input fields, it takes some time for validating it and showing error as invalid input even if the entered value is valid.

![Screenshot (148)](https://user-images.githubusercontent.com/94359116/177983436-85e550f0-74da-44df-a5fd-285cc3854c82.png)

**Comment by dilippitchika** [comment]
@ginjojohny this will be fixed in the next version, when the next version goes live can you please reconfigure the regex in the JSON form? It will work then

**Comment by ginjojohny** [comment]
@dilippitchika Thank you for the update.  Is it available in v1.7.10  ?

**Comment by dilippitchika** [comment]
Hey @ginjojohny it will be available in v1.7.11

**Comment by ginjojohny** [comment]
@dilippitchika  Ok Thanks for the update.
